native/proto/src/proto/expr.proto

```diff
@@ -86,6 +86,7 @@ message Expr {
     ArrayInsert array_insert = 59;
     BinaryExpr array_contains = 60;
     BinaryExpr array_remove = 61;
+    BinaryExpr array_join = 63;
```

Comment:
Why not use `62`?

Reply:
I have another PR for [array_intersect](https://github.com/apache/datafusion-comet/pull/1271/files#diff-80c3ba0a1bf291317df58867a22e3f9abe681a16f7a8a9ec4959f744a0a2d509R89) and it uses 62. I think array_intersect can go first